numpy/lib/npyio.py

```diff
@@ -1875,6 +1875,7 @@ def genfromtxt(fname, dtype=float, comments='#', delimiter=None,
       exception is raised).
     * Individual values are not stripped of spaces by default.
       When using a custom converter, make sure the function does remove spaces.
+    * Custom converters may receive unexpected values when ``dtype=None``.
```

Comment:
Actually, this bug happens even if the dtype is not None.
<details>
<summary> How to reproduce </summary>

This gives the same output that was seen in [the original problem](https://stackoverflow.com/questions/75783751/reading-csv-with-numpy-has-extra-row-of-1s)

```
import numpy as np

filename = 'b.csv'
h = ("", "Col1", "Col2", "Col3")

def col1_converter(v):
    print(f'col1_converter {v = }')
    return v

def col2_converter(v):
    print(f'col2_converter {v = }')
    return v

def col3_converter(v):
    print(f'col3_converter {v = }')
    return v

a = np.genfromtxt(
    filename,
    delimiter=',',
    names=True,
    dtype=[np.dtype('U8'), np.dtype('U8'), np.dtype('U2'), np.dtype('U2')],
    usecols=range(1, len(h)),
    converters={1: col1_converter, 2: col2_converter, 3: col3_converter},
    deletechars='',
)
print()
print(a)
```
</details>

Reply:
@rossbar Ping.

Reply:
Okay, I've loosened the text to resolve the inaccuracy. Thanks @MatteoRaso !